src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -231,4 +264,49 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 			}
 		}
 	}
+
+	/**
+	 * Instantiate modal for use as callout when inserting Link or Image into markdown.
+	 * @param calloutStyle Style of callout passed in to determine which callout is rendered.
+	 * Returns markup created after user enters values and submits the callout.
+	 */
+	private async createCallout(type: MarkdownButtonType, triggerElement: HTMLElement): Promise<ILinkCalloutDialogOptions> {
+		const triggerPosX = triggerElement.getBoundingClientRect().left;
+		const triggerPosY = triggerElement.getBoundingClientRect().top;
+		const triggerHeight = triggerElement.offsetHeight;
+		const triggerWidth = triggerElement.offsetWidth;
+		const dialogProperties = { xPos: triggerPosX, yPos: triggerPosY, width: triggerWidth, height: triggerHeight };
+		let calloutOptions;
+		/**
+		 * Width value here reflects designs for Notebook callouts.
+		 */
+		const width: DialogWidth = 452;
+
+		if (type === MarkdownButtonType.LINK_PREVIEW) {
+			const defaultLabel = this.getCurrentSelectionText();
+			this._linkCallout = this._instantiationService.createInstance(LinkCalloutDialog, this.insertLinkHeading, width, dialogProperties, defaultLabel);
+			this._linkCallout.render();
+			calloutOptions = await this._linkCallout.open();
+		}
+		return calloutOptions;
+	}
+
+	private getCurrentSelectionText(): string {
+		if (this.cellModel.currentMode === CellEditModes.WYSIWYG) {
+			return document.getSelection()?.toString() || '';
+		} else {
+			const nbEditor = this._notebookService.findNotebookEditor(this.cellModel?.notebookModel?.notebookUri);
```

Comment:
I don't like having this code live in the toolbar component. I thought about adding the ability to query for a cell's backing editor, but that only saves a couple of lines of code.

Having the cellModel handle the "text of what's currently selected" was another thing I thought about, but I'm not convinced that makes sense to live there either (and it's a bit niche). Any thoughts for where I can put this? Maybe a utility method or something?